### PR TITLE
Subroot quick - Allow the app to be installed below the document root

### PR DESCRIPTION
**Pull request by Natorator**
Here's the quick and dirty version - because it was quicker, of course!

In index.php, we define a global BASE_PATH constant, which is used in the Router route declarations, and in UrlModel to produce uri paths that are always correctly relative to the document root, and in the BusinessModel to make sure file uploads still work.

### CODE REVIEW

**Comment by sitilge** [comment]
Hi,

Thank you for your efforts. IMHO, there is no need to define a constant since a route that starts with `/` is already relative to the root. The same applies for the static files.

What concerns image uploads - the `publicPath` (probably this should be renamed to `imagePublicPath`), `imageDomain` and `imageDir` allows to store images under different path and domain. For example, I have a site `abc.com` that resides under `/var/www/abc.com` and administration site `xyz.com` under `var/www/xyz.com` with the `publicPath` of `/var/www/xyz.com/public` and `imageDir` of  `img/uploads` that corresponds to  `/var/www/xyz.com/public/img/uploads`.

**Comment by Natorator** [comment]
Let me clarify. 

As it stands, the app can be installed with the public/ folder serving as the document root of a website. eg: http://example.com/index.php . This pull request adjusts things so that curdle can be installed in subfolders of the website's document root. eg: http://example.com/admin/index.php or http://example.com/curdle/index.php . This can be done by way of a symlink to the public folder. 

The codebase will still work perfectly fine when installed on the document root - this just adds more flexibility to the uri you can access the application with.

**Comment by sitilge** [comment]
@Natorator, sorry could you rephrase the comment. I don't quite understand your point.

AFAIK, there are no problems installing curdle under existing project's document root. Just provide the proper path in your web server's config/hosts.

By the way, the curdle's index is under public/index.php (if it helps you somehow)

**Comment by Natorator** [comment]
Let's say I have other apps on my website. Those are accessible from the document through the web at http://example.com/ . I want to use curdle to edit the database of that application, but I don't want to use the public/ folder as my  document root. so I install curdle in a subdirectory of the website. I want to access curdle at http://example.com/curdle/public/ instead. 

As it stands, the code base won't work with that - It insists on having public/ be the configured document root of the webstie. 

This patch allows for that, without breaking sites that have them installed in the document root.

File: app/Views/Admin/Template.php [hunk 1]
```diff
@@ -10,8 +10,8 @@
         <meta content="" property="og:image">
         <meta content="A simple database administration tool for simple CRUD operations." property="og:description">
         <meta content="https://github.com/sitilge/curdle" property="og:url">
-        <link rel="stylesheet" href="/css/all.css">
-        <script src="/js/all.js"></script>
+        <link rel="stylesheet" href="<?=$url->admin()?>css/all.css">
+        <script src="<?=$url->admin()?>js/all.js"></script>
```
Comment (sitilge) [inline app/Views/Admin/Template.php]: Eh, could we agree on using `<?php echo ` (trying not to split hairs here, but still...)

**Comment by sitilge** [comment]
@Natorator I got your point, agree. But could you manage to use the fetched value from `Config` files instead of defining a constant?

**Comment by sitilge** [comment]
@Natorator I got your point, agree. But could you manage to use the fetched value from `Config` files instead of defining a constant?

File: app/Views/Admin/Template.php (hunk 1 above)
Comment (Natorator) [inline app/Views/Admin/Template.php]: works for me.

It may be worth noting that while short tags "<?" are much-maligned (for good reason), echo tags "<?=" are now universally available, regardless of the short_open_tags php.ini setting
http://php.net/manual/en/migration54.new-features.php

**Comment by Natorator** [comment]
Sure. 

So it looks like the Abimo Factory is a static singleton. So I guess just instantiating a "new" isntance inside the UrlModel (or wherever it's needed) serves the purpose of global accessibility, and allows us to avoid long dependency injection cascades. Then Factory then provides the config, which provides the base path variable. Does that sound good?

**Comment by Natorator** [comment]
Updated. 'base_uri_path' is now in App.php the function call there should take care of it dynamically, but coders can feel free to override it if they like.
